### PR TITLE
Runtime refactoring

### PR DESCRIPTION
**Pull request by moriaab**
This refactoring includes:
- unifying all gas related code into one file and separating gas in runtime into struct
- creating wasm engine module:
          - responsible for deployment and execution of contracts and it
          - holds the runtime
- remove `wasm_g` and all its contents
- removing `types` from deployment and execution of contracts since it was not used

### CODE REVIEW

File: enigma-runtime-t/src/gas.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+
+pub use pwasm_utils::{inject_gas_counter, rules};
+
+/// Wasm cost table
+pub struct WasmCosts {
+    /// Default opcode cost
+    pub regular: u32,
+    /// Div operations multiplier.
+    pub div: u32,
+    /// Div operations multiplier.
```
Comment (AvishaiW) [inline enigma-runtime-t/src/gas.rs:10]: one of these comments (or both) isn't accurate.

File: enigma-runtime-t/src/wasm_execution.rs [hunk 2]
```diff
@@ -1,184 +1,163 @@
...
+        Self::new(&deploy_bytecode, gas_limit, args, state, function_name, key)
     }
-    let wasm_costs = WasmCosts::default();
-    let contract_module = pwasm_utils::inject_gas_counter(deserialized_module, &gas_rules(&wasm_costs))?;
-    let limited_module = pwasm_utils::stack_height::inject_limiter(contract_module, wasm_costs.max_stack_height)?;
 
-    let module = wasmi::Module::from_parity_wasm_module(limited_module)?;
-    Ok(Box::new(module))
-}
+    pub fn new_compute(code: &[u8], gas_limit: u64, args: Vec<u8>, state: ContractState, function_name: String,key: StateKey) -> Result<WasmEngine, EnclaveError>{
```
Comment (AvishaiW) [inline enigma-runtime-t/src/wasm_execution.rs:38]: this function seems irrelevant, but whatever you think..

File: enigma-runtime-t/src/wasm_execution.rs [hunk 3]
```diff
@@ -1,184 +1,163 @@
...
..., 2, 24, 32, 1, 32, 2, 32, 0, 65, 28, 106, 40, 2, 0, 40, 2, 12, 17, 1, 0, 33, 4, 11, 32, 3, 65, 16, 106, 36, 0, 32, 4, 11, 126, 1, 2, 127, 2, 64, 32, 0, 40, 2, 4, 34, 2, 32, 0, 40, 2, 8, 34, 3, 107, 32, 1, 79, 13, 0, 2, 64, 2, 64, 32, 3, 32, 1, 106, 34, 1, 32, 3, 73, 13, 0, 32, 2, 65, 1, 116, 34, 3, 32, 1, 32, 1, 32, 3, 73, 27, 34, 1, 65, 0, 72, 13, 0, 2, 64, 2, 64, 32, 2, 69, 13, 0, 32, 0, 40, 2, 0, 32, 1, 16, 42, 34, 2, 69, 13, 1, 12, 3, 11, 32, 1, 65, 1, 16, 43, 34, 2, 13, 2, 11, 0, 0, 11, 16, 36, 0, 11, 32, 0, 32, 2, 54, 2, 0, 32, 0, 65, 4, 106, 32, 1, 54, 2, 0, 11, 11, 230, 5, 1, 8, 127, 65, 0, 33, 2, 2, 64, 2, 64, 2, 64, 2, 64, 2, 64, 32, 1, 65, 191, 127, 75, 13, 0, 65, 16, 32, 1, 65, 11, 106, 65, 120, 113, 32, 1, 65, 11, 73, 27, 33, 3, 32, 0, 65, 124, 106, 34, 4, 40, 2, 0, 34, 5, 65, 120, 113, 33, 6, 2, 64, 2, 64, 2, 64, 2, 64, 2, 64, 32, 5, 65, 3, 113, 69, 13, 0, 32, 0, 65, 120, 106, 34, 7, 32, 6, 106, 33, 8, 32, 6, 32, 3, 79, 13, 1, 65, 0, 40, 2, 192, 147, 65, 32, 8, 70, 13, 2, 65, 0, 40, 2, 188, 147, 65, 32, 8, 70, 13, 3, 32, 8, 40, 2, 4, 34, 5, 65, 2, 113, 13, 4, 32, 5, 65, 120, 113, 34, 9, 32, 6, 106, 34, 6, 32, 3, 73, 13, 4, 32, 6, 32, 3, 107, 33, 1, 32, 9, 65, 255, 1,
```
Comment (AvishaiW) [inline enigma-runtime-t/src/wasm_execution.rs:149]: I know this is not related but is it possible to move this to an external file. it doesn't seem so efficient here.

**Review by AvishaiW — APPROVED** [review]
WOW, Great Job! The code is much cleaner and it's very efficient!

File: enigma-runtime-t/src/wasm_execution.rs (hunk 3 above)
Comment (moriaab) [reply enigma-runtime-t/src/wasm_execution.rs:149]: Good point, will do this.
